src/rdsn/src/common/backup_restore_common.cpp

```diff
@@ -36,5 +44,41 @@ const std::string backup_restore_constant::BACKUP_ID("restore.backup_id");
 const std::string backup_restore_constant::SKIP_BAD_PARTITION("restore.skip_bad_partition");
 const std::string backup_restore_constant::RESTORE_PATH("restore.restore_path");
 
+std::string get_backup_root(const std::string &backup_root,
+                            const std::string &user_defined_root_path)
+{
+    if (user_defined_root_path.empty()) {
+        return backup_root;
+    }
+    return utils::filesystem::path_combine(user_defined_root_path, backup_root);
```

Comment:
if user specify root path, we should return the value, but not combine `system` + `user` path

Reply:
If user specify root_path, we will return `root_path/backup_root`, otherwise return backup_root, backup_root is actually the cluster name.